import os

from ament_index_python.packages import get_package_share_directory


from launch import LaunchDescription
from launch.actions import IncludeLaunchDescription
from launch.launch_description_sources import PythonLaunchDescriptionSource

from launch_ros.actions import Node

def generate_launch_description():
    spawn_drive = Node(package="drive",            
        executable="drive_node",
        name="drive_node",
        parameters=[{"use_sim_time": True}]
    )
    
    spawn_left_motor = Node(package = "motor_control",
    executable = "sparkmax_control_node",
    name = "sparkmax_control_node",
    parameters=[{"use_sim_time": True},
                {"motor_name": "left_motor"},
                {"can_interface": "can1"},
                {"can_id": 11},
                {"control_topic": "/front_left/control"},
                {"status_topic": "/front_left/status"},
                {"health_topic": "/front_left/health"}],
    arguments=["--ros-args",
               "-r",
               "__node:=left_motor_controller"]
    )

    spawn_right_motor = Node(package = "motor_control",
    executable = "sparkmax_control_node",
    name = "sparkmax_control_node",
    parameters=[{"use_sim_time": True},
                {"motor_name": "front_right"},
                {"can_interface": "can1"},
                {"can_id": 10},
                {"control_topic": "/front_right/control"},
                {"status_topic": "/front_right/status"},
                {"health_topic": "/front_right/health"}],
    arguments=["--ros-args",
               "-r",
               "__node:=right_motor_controller"]
    )

    return LaunchDescription([
        spawn_drive,
        spawn_left_motor,
        spawn_right_motor
    ])

